File: src/TVSeries.java
/**
 * TVSeries is a subclass of Media. The objects of TVSeries contain information about a TV Series.
 * Title, series, episode, publish year, rating, genre can be stored in to the objects.
 * 
 * @author      Juha Hirvasniemi <jhirvasn@paju.oulu.fi>, Tapio Korvala <korvatap@paju.oulu.fi>
 * @version     1.0
 * @since       2013-12-30          (the version of the package this class was first added to)
 */

public class TVSeries extends Media {

	static final long serialVersionUID = 53L;
	String episode;
	String series;
	private int id;
	
	/**
	 * Constructor for TVSeries class which can store the information about TV series
	 * without using any setters.
	 * 
	 * @param title		title for the tv series.
	 * @param series 	season name for the series or a number.
	 * @param episode 	name of the episode or a number.
	 * @param publishYear the year of the current episode was produced.
	 * @param rating 	the rating of the episode, 1-10.
	 * @param genre 	the genre for the tv series e.x. Action.
	 */
	public TVSeries(String title, String series, String episode, int publishYear, int rating, String genre) {
		super.increaseId();
		this.id = super.getId();
		super.setTitle(title);
		this.series = series;
		this.episode = episode;
		super.setPublishYear(publishYear);
		super.setRating(rating);
		super.setGenre(genre);
	}
	
	/**
	 * Returns id of the created object.
	 * 
	 * @return id	id of the current object.
	 */
	public int getId() {
		return id;
	}
	
	/**
	 * Sets the episode name to the wanted value.
	 * 	
	 * @param e		value to set as episode name.
	 */
	public void setEpisode(String e) {
		this.episode = e;
	}
	
	/**
	 * Returns the name of the episode.
	 * 
	 * @return		name of the episode.
	 */
	public String getEpisode() {
		return this.episode;
	}
	
	/**
	 * Sets the name of the season.
	 * 
	 * @param s		value to be set as name of the season.
	 */
	public void setSeries(String s) {
		this.series = s;
	}
	
	/**
	 * Returns the name of the season.
	 * 
	 * @return		name of the season.
	 */
	public String getSeries() {
		return series;
	}
	
	public void print() {
        System.out.println("-------------");
        System.out.println("title: " + getTitle() + ", Series: " + getSeries() + ", Episode: " + getEpisode() +  ", PublishYear: " + getPublishYear() + ", Rating: " + getRating() + ", Genre: " + getGenre());
        System.out.println("-------------");
    }
	
	/**
	 * Returns all the values of the TV Series as string array.
	 * 
	 * @return		values of tv series.
	 */
	public String[] getRow() {
		String [] row = { Integer.toString(getId()), super.getTitle(), series, episode, Integer.toString(super.getPublishYear()), Integer.toString(super.getRating()), super.getGenre() };
		return row;
	}
}

File: web/TVSeries.php
<?php

class TVSeries extends Media {

    private $episode;
    private $season;
    private $id;

    public function __construct() {
        $arguments = func_get_args();
        $numOfArgs = func_num_args();
        if (method_exists ($this, $function = '__construct' .$numOfArgs) ) {
            call_user_func_array (array ($this, $function), $arguments);
        } 
    }
    public function __construct6($title, $season, $episode, $publishYear, $rating, $genre) {
        parent::setTitle($title);
        $this->season = $season;
        $this->episode = $episode;
        parent::setPublishYear($publishYear);
        parent::setRating($rating);
        parent::setGenre($genre);
        parent::increaseId();
        $this->id = parent::getId();
    }

    public function setEpisode($episode) {
        $this->episode = $episode;
    }
    
    public function getEpisode() {
        return $this->episode;
    }
    
    public function setSeason($season) {
        $this->season = $season;
    }
    
    public function getSeason() {
        return $this->season;
    }
    
    public function id() {
        return $this->id;
    }
    
    public function info() {
        echo "<p id='teksti'>-------------<br/></p>";
        echo "<p id='teksti'>ID: " .$this->id() .", Title: " .parent::getTitle() .", Season: " .$this->getSeason() .", Episode: " .$this->getEpisode() .", PublishYear: " .parent::getPublishYear() .", Rating: " .parent::getRating() .", Genre: " .parent::getGenre() ." <a href='?link=Modify&id=$this->id()&type=TVSeries'>Modify</a> ". "<a href=?link=Delete&id=$this->id()&type=TVSeries>Delete</a>"."<br/></p>";
        echo "<p id='teksti'>-------------<br/></p>";
    }
}
?>